xlators/mgmt/glusterd/src/glusterd-snapshot-utils.c

```diff
@@ -248,11 +248,12 @@ glusterd_snap_volinfo_restore(dict_t *dict, dict_t *rsp_dict,
             /* To use generic functions from the plugin */
             glusterd_snapshot_plugin_by_name(snap_volinfo->snap_plugin,
                                              &snap_ops);
-
-            snap_ops->brick_path(snap_mount_dir, brickinfo->origin_path, 0,
-                                 snap_volinfo->snapshot->snapname,
-                                 snap_volinfo->volname, brickinfo->mount_dir,
-                                 brick_count - 1, new_brickinfo, 1);
+            if (snap_ops) {
```

Comment:
@aravindavk can you check if that's the right thing to do ? shouldn't we exit this function with an error if `snap_ops` is `NULL` ?

Reply:
I think it will not be ever NULL.

Reply:
It will be `NULL` if brick is on a plain XFS mount, for example. However, if we are sure that this won't happen because there's some code path before reaching this point that has already verified that, we shouldn't check it here. At most we should do a `GF_ASSERT()`.